src/Sylius/Behat/Context/Ui/Shop/ProductReviewContext.php

```diff
@@ -0,0 +1,196 @@
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\NotificationType;
+use Sylius\Behat\Page\Shop\ProductReview\CreatePageInterface;
+use Sylius\Behat\Service\NotificationCheckerInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class ProductReviewContext implements Context
+{
+    /**
+     * @var CreatePageInterface
+     */
+    private $createPage;
+
+    /**
+     * @var NotificationCheckerInterface
+     */
+    private $notificationChecker;
+
+    /**
+     * @param CreatePageInterface $createPage
+     * @param NotificationCheckerInterface $notificationChecker
+     */
+    public function __construct(
+        CreatePageInterface $createPage,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->createPage = $createPage;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @Given I want to review product :product
+     */
+    public function iWantToReviewProduct(ProductInterface $product)
+    {
+        $this->createPage->open(['slug' => $product->getSlug()]);
+    }
+
+    /**
+     * @When I leave a comment :comment as :author
+     * @When I leave a comment :comment, titled :title
+     * @When I leave a comment :comment, titled :title as :author
+     * @When I leave a review titled :title as :author
+     */
+    public function iLeaveACommentTitled($comment = null, $title = null, $author = null)
+    {
+        $this->createPage->titleReview($title);
+        $this->createPage->setComment($comment);
+
+        if (null !== $author) {
+            $this->createPage->setAuthor($author);
+        }
+    }
+
+    /**
+     * @When I title it with very long title
+     */
+    public function iTitleItWithVeryLongTitle()
+    {
+        $this->createPage->titleReview($this->getVeryLongTitle());
+    }
+
+    /**
+     * @When I rate it with :rate point(s)
+     */
+    public function iRateItWithPoints($rate)
+    {
+        $this->createPage->rateReview($rate);
+        $this->createPage->submitReview();
+    }
+
+    /**
+     * @Then I should be notified that my review is waiting for the acceptation
+     */
+    public function iShouldBeNotifiedThatMyReviewIsWaitingForTheAcceptation()
+    {
+        $this->notificationChecker->checkNotification('Your review is waiting for the acceptation.', NotificationType::success());
+    }
+
+    /**
+     * @When I do not rate it
+     */
+    public function iDoNotRateIt()
+    {
+        $this->createPage->submitReview();
+    }
+
+    /**
+     * @Then I should be notified that I must check review rating
+     */
+    public function iShouldBeNotifiedThatIMustCheckReviewRating()
+    {
+        Assert::same(
+            $this->createPage->getRateValidationMessage(),
+            'You must check review rating.',
+            'There should be rate validation error, but there is not.'
```

Comment:
I think everywhere else we follow:
```
Assert::same(
    'You must check review rating.',
    $this->createPage->getRateValidationMessage(),
    'There should be rate validation error, but there is not.'
);
```
To separate the expected message from validation message. IMO it's more readable.